internal/widget/termgrid.go

```diff
@@ -40,7 +40,7 @@ func NewTermGrid() *TermGrid {
 // We update our blinking status and then call the TextGrid we extended to refresh too.
 func (t *TermGrid) Refresh() {
 	t.refreshBlink(false)
-	t.TextGrid.Refresh()
```

Comment:
I'm not sure about the others but this one seems wrong.
Refreshing a widget should happen on the main thread - so the cause is further up the call tree.